File: tests/statistical_tests/spot_checks/shapiro-wilk.test.ts
import { stats } from "../../../src/dataframe/mod.ts";

Deno.test("Shapiro-Wilk Normality Test Spot Check", () => {
  console.log("\n📊 Shapiro-Wilk Normality Test");
  console.log("-".repeat(80));

  const data = [2.5, 3.1, 2.8, 3.3, 2.9, 3.0, 2.7, 3.2];

  const result = stats.test.normality.shapiroWilk({ data });

  console.log("Data:", data);
  console.log("W-statistic:", result.test_statistic.value);
  console.log("p-value:", result.p_value);
});

console.log("\n" + "=".repeat(80));
console.log("SHAPIRO-WILK NORMALITY TEST SPOT CHECK");
console.log("=".repeat(80));

File: tests/statistical_tests/spot_checks/shapiro-wilk.test.R
#!/usr/bin/env Rscript

cat("\n📊 Shapiro-Wilk Normality Test\n")
cat(paste(rep("-", 80), collapse = ""), "\n")

data <- c(2.5, 3.1, 2.8, 3.3, 2.9, 3.0, 2.7, 3.2)

# Shapiro-Wilk test
result <- shapiro.test(data)

cat("Data:", data, "\n")
cat("W-statistic:", result$statistic, "\n")
cat("p-value:", result$p.value, "\n")

cat("\n", paste(rep("=", 80), collapse = ""), "\n")
cat("SHAPIRO-WILK NORMALITY TEST SPOT CHECK\n")
cat(paste(rep("=", 80), collapse = ""), "\n")
